Postal/Postal.swift

```diff
@@ -268,9 +267,17 @@ public extension Postal {
 
 private extension Postal {
     
-    func doAsync<T, E>(_ f: @escaping () throws -> T, completion: @escaping (Result<T, E>) -> Void) {
+    func doAsync<T>(_ f: @escaping () throws -> T, completion: @escaping (Result<T, PostalError>) -> Void) {
         queue.addOperation {
-            let result = Result<T, E>(attempt: f)
+            let result: Result<T, PostalError>
+            do {
+                let success = try f()
+                result = .success(success)
+            } catch let error as PostalError {
+                result = .failure(error)
+            } catch {
+                result = .failure(.undefined)
+            }
```

Comment:
This needed to get a bit more complicated since as far as this function is concerned, `f()` throws type `Error`, not `E`. I'm not sure if there's a way to specify the type of error that can be thrown from a throwing function.

Reply:
If you want, we can even put `E` back, and catch all errors of type `E` like so:

```
catch let error as E
```

But I figured I'd make it simpler since we seem to only be throwing `PostalError`s.